cmd/init.go

```diff
@@ -0,0 +1,187 @@
+		Long:    ``,
+		Example: ``,
+		Annotations: map[string]string{
+			"group": "dataset",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			o.Complete(f)
+			return o.Run()
+		},
+	}
+
+	cmd.PersistentFlags().StringVar(&o.Name, "name", "", "name of the dataset")
+	cmd.PersistentFlags().StringVar(&o.Format, "format", "", "format of dataset")
+
+	return cmd
+}
+
+// InitOptions encapsulates state for the `init` command
+type InitOptions struct {
+	ioes.IOStreams
+
+	Name            string
+	Format          string
+	DatasetRequests *lib.DatasetRequests
+}
+
+// Complete completes a dataset reference
+func (o *InitOptions) Complete(f Factory) (err error) {
+	o.DatasetRequests, err = f.DatasetRequests()
+	return err
+}
+
+// Run executes the `init` command
+func (o *InitOptions) Run() (err error) {
+	if _, err := os.Stat(QriRefFilename); !os.IsNotExist(err) {
+		return fmt.Errorf("working directory is already linked, .qri-ref exists")
+	}
+	if _, err := os.Stat("meta.json"); !os.IsNotExist(err) {
+		// TODO(dlong): Instead, import the meta.json file for the new dataset
+		return fmt.Errorf("cannot initialize new dataset, meta.json exists")
+	}
+	if _, err := os.Stat("schema.json"); !os.IsNotExist(err) {
+		// TODO(dlong): Instead, import the schema.json file for the new dataset
+		return fmt.Errorf("cannot initialize new dataset, schema.json exists")
+	}
+	if _, err := os.Stat("body.csv"); !os.IsNotExist(err) {
+		// TODO(dlong): Instead, import the body.csv file for the new dataset
+		return fmt.Errorf("cannot initialize new dataset, body.csv exists")
+	}
+	if _, err := os.Stat("body.json"); !os.IsNotExist(err) {
+		// TODO(dlong): Instead, import the body.json file for the new dataset
+		return fmt.Errorf("cannot initialize new dataset, body.json exists")
+	}
+
+	// Suggestion for the dataset name defaults to the name of the current directory.
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	suggestDataset := filepath.Base(pwd)
+
+	// Process flags for inputs, prompt for any that were not provided.
+	var dsName, dsFormat string
+	if o.Name != "" {
+		dsName = o.Name
+	} else {
+		dsName = inputText(o.ErrOut, o.In, "Name of new dataset", suggestDataset)
+	}
+	if o.Format != "" {
+		dsFormat = o.Format
+	} else {
+		dsFormat = inputText(o.ErrOut, o.In, "Format of dataset, csv or json", "csv")
+	}
+
+	ref := fmt.Sprintf("me/%s", dsName)
+
+	// Validate dataset name. The `init` command must only be used for creating new datasets.
+	// Make sure a dataset with this name does not exist in your repo.
+	p := lib.GetParams{
+		Path:     ref,
+		Selector: "",
+	}
+	res := lib.GetResult{}
+	if err = o.DatasetRequests.Get(&p, &res); err == nil {
+		// TODO(dlong): Tell user to use `checkout` if the dataset already exists in their repo?
+		return fmt.Errorf("a dataset with the name %s already exists in your repo", ref)
+	}
+
+	// Validate dataset format
+	if dsFormat != "csv" && dsFormat != "json" {
```

Comment:
we're so close to having a nice method for checking these strings in dataset: https://godoc.org/github.com/qri-io/dataset#SupportedDataFormats I'll PR in a new package function for checking if a string is a supported Data format